jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/kafka/producer/InterceptingKafkaTemplate.java

```diff
@@ -52,6 +83,42 @@ protected CompletableFuture<SendResult<K, V>> doSend(
       return new CompletableFuture<>();
     }
 
+    // For String values, extract the originIp from the JSON metadata if it is
+    // present
+    String originIp = null;
+    if (producerRecord.value() instanceof String) {
+      String stringValue = (String) producerRecord.value();
```

Comment:
suggestion(non-blocking): you can reduce some redundancy if you use the following syntax
```suggestion
    if (producerRecord.value() instanceof String stringValue) {
```

Reply:
Interesting, I didn't actually know you could do this!

Reply:
Addressed